File: app/src/main/java/com/fortitude/shamsulkarim/ieltsfordory/data/repository/AudioRepository.java
package com.fortitude.shamsulkarim.ieltsfordory.data.repository;

import java.io.File;

public interface AudioRepository {
    interface Callback {
        void onAudioReady(File audioFile);

        void onError(Exception e);
    }

    void downloadAudio(String wordName, Callback callback);
}

File: app/src/main/java/com/fortitude/shamsulkarim/ieltsfordory/domain/media/AudioRepository.kt
package com.fortitude.shamsulkarim.ieltsfordory.domain.media

import com.fortitude.shamsulkarim.ieltsfordory.domain.media.model.AudioData

interface AudioRepository {
    interface Callback {
        fun onSuccess(data: AudioData)
        fun onError(e: Exception)
    }

    fun downloadAudio(wordName: String, callback: Callback)
}



